test/bot_test.exs

```diff
@@ -8,32 +10,35 @@ defmodule Slack.BotTest do
   @rtm %{
     url: "http://example.com",
     self: %{name: "fake"},
-    team: %{name: "Foo"},
-    bots: [%{id: "123"}],
-    channels: [%{id: "123"}],
-    groups: [%{id: "123"}],
-    users: [%{id: "123"}],
-    ims: [%{id: "123"}]
+    team: %{name: "Foo"}
   }
 
   test "init formats rtm results properly" do
-    {:reconnect, %{slack: slack, bot_handler: bot_handler}} =
-      Slack.Bot.init(%{
-        bot_handler: Bot,
-        rtm: @rtm,
-        client: FakeWebsocketClient,
-        token: "ABC",
-        initial_state: nil
-      })
-
-    assert bot_handler == Bot
-    assert slack.me.name == "fake"
-    assert slack.team.name == "Foo"
-    assert slack.bots == %{"123" => %{id: "123"}}
-    assert slack.channels == %{"123" => %{id: "123"}}
-    assert slack.groups == %{"123" => %{id: "123"}}
-    assert slack.users == %{"123" => %{id: "123"}}
-    assert slack.ims == %{"123" => %{id: "123"}}
+    with_mocks([
```

Comment:
I've kept it as somewhat of a personal goal to keep `mock` out of the project. I'm happy to take it on myself, but it'd be great to see the fake http client return these instead of mocking.

Reply:
I tried for a few hours on Saturday and this was the only way I could think of to mock out the external Web calls